appinventor/appengine/src/com/google/appinventor/server/storage/ObjectifyStorageIo.java

```diff
@@ -1813,6 +1854,22 @@ public void run(Objectify datastore) {
       } else if (fileData.isBlob) {
         try {
           result.t = getBlobstoreBytes(fileData.blobstorePath);
+          // Time to consider upgrading this file if we are moving to GCS
+          // Note: We only run if we have at least 5 seconds of runtime left in the request
+          long timeRemaining = ApiProxy.getCurrentEnvironment().getRemainingMillis();
+          if (conversionEnabled && useGcs &&
+            (timeRemaining > 5000)) {
```

Comment:
I'm not sure I know what you mean. We must fetch the result (result.t = getBlobstoreBytes...) regardless of the time remaining, we are in a read call. But if we are low on time, we don't bother with the re-write (which will do the upgrade).